test/format-test.cc

```diff
@@ -1344,16 +1344,82 @@ TEST(format_test, format_double) {
   EXPECT_EQ(fmt::format("{:f}", 392.65), "392.650000");
   EXPECT_EQ(fmt::format("{:F}", 392.65), "392.650000");
   EXPECT_EQ(fmt::format("{:L}", 42.0), "42");
+  EXPECT_EQ(fmt::format("{:24a}", 4.2f), "           0x1.0cccccp+2");
   EXPECT_EQ(fmt::format("{:24a}", 4.2), "    0x1.0cccccccccccdp+2");
   EXPECT_EQ(fmt::format("{:<24a}", 4.2), "0x1.0cccccccccccdp+2    ");
   EXPECT_EQ(fmt::format("{0:e}", 392.65), "3.926500e+02");
   EXPECT_EQ(fmt::format("{0:E}", 392.65), "3.926500E+02");
   EXPECT_EQ(fmt::format("{0:+010.4g}", 392.65), "+0000392.6");
   char buffer[buffer_size];
+  double d = 0;
+
+#if FMT_CPLUSPLUS >= 201703L
+  d = 0x1.ffffffffffp+2;
+  safe_sprintf(buffer, "%.*a", 10, d);
+  EXPECT_EQ(fmt::format("{:.10a}", d), buffer);
+  safe_sprintf(buffer, "%.*a", 9, d);
+  EXPECT_EQ(fmt::format("{:.9a}", d), buffer);
+
+  if (std::numeric_limits<long double>::digits == 64) {
+    auto ld = 0xf.ffffffffffp-3l;
+    safe_sprintf(buffer, "%La", ld);
+    EXPECT_EQ(fmt::format("{:a}", ld), buffer);
+    safe_sprintf(buffer, "%.*La", 10, ld);
+    EXPECT_EQ(fmt::format("{:.10a}", ld), buffer);
+    safe_sprintf(buffer, "%.*La", 9, ld);
+    EXPECT_EQ(fmt::format("{:.9a}", ld), buffer);
+  }
+#endif
+
+  d = (std::numeric_limits<double>::min)();
+  safe_sprintf(buffer, "%a", d);
+  std::string d_min_outputs[] = {
+      "0x1p-1022",               // UNIX
+      "0x1.0000000000000p-1022"  // Windows
+  };
+  EXPECT_THAT(d_min_outputs, testing::Contains(buffer));
```

Comment:
I don't think we need to test sprintf output. Let's remove this check here and in similar places below.

Reply:
Done